addons/account/models/account_report.py

```diff
@@ -338,6 +338,27 @@ def _inverse_aggregation_formula(self):
     def _inverse_account_codes_formula(self):
         self._create_report_expression(engine='account_codes')
 
+    @api.ondelete(at_uninstall=False)
+    def _unlink_archive_used_tags(self):
+        """
+        Manages unlink or archive of tax_tags when a report.line is deleted.
+        If it's still in use either on amls or on other expressions
+        """
+        for expression in self.expression_ids:
+            if expression.engine == 'tax_tags':
+                country = expression.report_line_id.report_id.country_id
+                tags = self.env['account.account.tag']._get_tax_tags(expression.formula, country.id)
+                for tag in tags:
+                    expression_using_tag = self.env['account.report.expression'].sudo().search([('engine', '=', 'tax_tags'), ('formula', '=', tag.name)], limit=1)
```

Comment:
tag.name contains the sign, and the formula does not ; your search will always return no result. You need to remove the first character of tag.name (in this case, it will always be the sign).

You probably should add a test for this, since it does not seem to break any existing one (it seems weird to me though; check that we don't have something that should check it and actually does not, maybe you'd just need to adapt it).